From fa15f7d65ad76872f9019f34dd2a1db0f678b0f9 Mon Sep 17 00:00:00 2001
From: Mike Fleetwood <mike.fleetwood@googlemail.com>
Date: Sat, 1 Oct 2016 16:40:19 +0100
Subject: [PATCH 48/53] libparted: Fix to report success when setting lvm flag
 on bsd table

bsd_partition_set_flag() was falling through in the lvm flag case and
returning failure.  Fix this by adding missing return 1 (success) like
for the other flags.

Found as a result of this bug reported by Timo Riikonen:
https://bugzilla.gnome.org/show_bug.cgi?id=769831

Signed-off-by: Brian C. Lane <bcl@redhat.com>
---
 libparted/labels/bsd.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/libparted/labels/bsd.c b/libparted/labels/bsd.c
index a8525a4..23daea8 100644
--- a/libparted/labels/bsd.c
+++ b/libparted/labels/bsd.c
@@ -488,6 +488,7 @@ bsd_partition_set_flag (PedPartition* part, PedPartitionFlag flag, int state)
 				bsd_data->raid = 0;
 			}
 			bsd_data->lvm = state;
+			return 1;
 		default:
 			;
 	}
-- 
2.7.4

